Sources/SwiftIfConfig/ConfiguredRegions.swift

```diff
@@ -13,6 +13,90 @@
 import SwiftDiagnostics
 import SwiftSyntax
 
+/// Describes all of the #if/#elseif/#else clauses within the given syntax node,
+/// indicating their active state. This operation will recurse into all
+/// clauses to indicate regions of active / inactive / unparsed code.
+///
+/// For example, given code like the following:
+/// #if DEBUG
+///   #if A
+///     func f()
+///   #elseif B
+///     func g()
+///   #elseif compiler(>= 12.0)
+///   please print the number after 41
+///   #endif
+/// #else
+/// #endif
+///
+/// If the configuration options `DEBUG` and `B` are provided, but `A` is not,
+/// and the compiler version is less than 12.0, the results will be contain:
+///   - Active region for the `#if DEBUG`.
+///   - Inactive region for the `#if A`.
+///   - Active region for the `#elseif B`.
+///   - Unparsed region for the `#elseif compiler(>= 12.0)`.
+///   - Inactive region for the final `#else`.
+public struct ConfiguredRegions {
+  let regions: [Element]
```

Comment:
Defining this as `Element` where `Element` is only defined in the conformance to `RandomAccessCollection` below seems a little confusing to me (I wasn’t able to figure it out on GitHub alone). I think it would be clearer to explicitly spell out the tuple type explicitly here 

```swift
let regions: [(IfConfigClauseSyntax, IfConfigRegionState)]
```

Reply:
Sure